apps/frontend/src/types/api.types.ts

```diff
@@ -9,140 +11,176 @@ export enum httpMethod {
   [ApiRequests.GET_CLIENTS]: httpMethod.GET,
   [ApiRequests.GET_CLIENT_BY_ID]: httpMethod.GET,
   [ApiRequests.UPDATE_CLIENT]: httpMethod.PATCH,
   [ApiRequests.CREATE_CLIENT]: httpMethod.POST,
-  /** Contact request methods */
+  /* Contact request methods */
   [ApiRequests.GET_CONTACTS]: httpMethod.GET,
   [ApiRequests.CREATE_CONTACT]: httpMethod.POST,
   [ApiRequests.UPDATE_CONTACT]: httpMethod.PATCH,
-  /** Opportunity request methods */
+  /* Opportunity request methods */
   [ApiRequests.GET_OPPORTUNITIES]: httpMethod.GET,
   [ApiRequests.GET_OPPORTUNITY_BY_ID]: httpMethod.GET,
   [ApiRequests.UPDATE_OPPORTUNITY]: httpMethod.PATCH,
   [ApiRequests.CREATE_OPPORTUNITY]: httpMethod.POST,
   [ApiRequests.GET_OPPORTUNITY_BY_CLIENT_ID]: httpMethod.GET,
+  /* Follow-up request methods */
+  [ApiRequests.GET_FOLLOW_UPS]: httpMethod.GET,
+  [ApiRequests.CREATE_FOLLOW_UP]: httpMethod.POST,
+  [ApiRequests.UPDATE_FOLLOW_UP]: httpMethod.PATCH,
 };
 
+// Body types for API requests
 export type ApiRequestBodyType = {
   [ApiRequests.CHECK_HEALTH]: void;
-  /** Client body types */
+  /* Client body types */
   [ApiRequests.GET_CLIENTS]: void;
   [ApiRequests.GET_CLIENT_BY_ID]: void;
   [ApiRequests.UPDATE_CLIENT]: Partial<ClientType>;
   [ApiRequests.CREATE_CLIENT]: Partial<ClientType>;
-  /** Contact body types */
+  /* Contact body types */
   [ApiRequests.GET_CONTACTS]: void;
   [ApiRequests.UPDATE_CONTACT]: ContactType;
   [ApiRequests.CREATE_CONTACT]: Partial<ContactType>;
-  /** Opportunity body types */
+  /* Opportunity body types */
   [ApiRequests.GET_OPPORTUNITIES]: void;
   [ApiRequests.GET_OPPORTUNITY_BY_ID]: void;
   [ApiRequests.UPDATE_OPPORTUNITY]: Partial<OpportunityType>;
   [ApiRequests.CREATE_OPPORTUNITY]: OpportunityType;
   [ApiRequests.GET_OPPORTUNITY_BY_CLIENT_ID]: void;
+  /* Follow-up body types */
+  [ApiRequests.GET_FOLLOW_UPS]: void;
+  [ApiRequests.CREATE_FOLLOW_UP]: FollowUpType;
+  [ApiRequests.UPDATE_FOLLOW_UP]: Partial<FollowUpType>;
 };
 
+// Query parameter types for API requests
 export type ApiRequestQueryType = {
   [ApiRequests.CHECK_HEALTH]: void;
-  /** Client query types */
+  /* Client query types */
   [ApiRequests.GET_CLIENTS]: paginationType & Partial<ClientType>;
   [ApiRequests.GET_CLIENT_BY_ID]: void;
   [ApiRequests.UPDATE_CLIENT]: void;
   [ApiRequests.CREATE_CLIENT]: void;
-  /** Contact query types */
+  /* Contact query types */
   [ApiRequests.GET_CONTACTS]: { clientId?: string } & Partial<ContactType>;
   [ApiRequests.CREATE_CONTACT]: void;
   [ApiRequests.UPDATE_CONTACT]: void;
-  /** Opportunity query types */
+  /* Opportunity query types */
   [ApiRequests.GET_OPPORTUNITIES]: paginationType & Partial<OpportunityType>;
   [ApiRequests.GET_OPPORTUNITY_BY_ID]: void;
   [ApiRequests.UPDATE_OPPORTUNITY]: void;
   [ApiRequests.CREATE_OPPORTUNITY]: void;
   [ApiRequests.GET_OPPORTUNITY_BY_CLIENT_ID]: void;
+  /* Follow-up query types */
+  [ApiRequests.GET_FOLLOW_UPS]: { opportunityId?: string };
```

Comment:
```suggestion
  [ApiRequests.GET_FOLLOW_UPS]: Partial<FollowUpType>
```

Asi podemos hacer mas filtrados